src/main/ipc.js

```diff
@@ -117,8 +124,28 @@ function init (cwd, state) {
     dc.updateSettings(saved)
   })
 
-  ipc.on('retrieveAdvancedSettingsObj', (e, ...args) => {
-    windows.main.send('retrieveAdvancedSettingsObjResp', dc.getAdvancedSettings())
+  ipc.on('updateCredentials', (e, credentials) => {
+    var dir = path.join(os.tmpdir(), Date.now().toString())
+    var tmp = new DeltaChat(dir, state.saved)
+
+    tmp.on('DC_EVENT_CONFIGURE_PROGRESS', function (data1) {
+      if (Number(data1) === 0) { // login failed
+        windows.main.send('error', 'Login failed')
+      }
+    })
+
+    function fakeRender () {
+      var json = dc.render()
+      var tmpJson = tmp.render()
+      json.configuring = tmpJson.configuring
+      windows.main.send('render', json)
+      if (tmpJson.ready) {
+        dc.login(credentials, render, txCoreStrings())
+        windows.main.send('success', 'Configuration success!')
+      }
+    }
+
+    tmp.login(credentials, fakeRender, txCoreStrings())
```

Comment:
I think we need some cleanup as well, the dc context inside the temporary `DeltaChat()` controller should be closed before (or after) `window.main.send()` has been called.

Reply:
ok